### PR TITLE
Port build changes from master to dev/release/2.0.0

### PR DESCRIPTION
**Pull request by eerhardt**
In support of removing the https://github.com/dotnet/source-build patches for core-setup, I am porting the build changes that were made in master after release/2.0.0 was forked.  This will allow the source-build work to progress.

The first 9 commits are straight cherry-picks from master.  The last 2 commits I had to do some manual fix ups.

/cc @dotnet/source-build-contrib

### CODE REVIEW

File: src/managed/CommonManaged.props [hunk 1]
```diff
@@ -15,15 +15,6 @@
     <PackageThirdPartyNoticesFile>$(RepoRoot)THIRD-PARTY-NOTICES.TXT</PackageThirdPartyNoticesFile>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(StabilizePackageVersion)' != 'true'">
```
Comment (weshaggard) [inline src/managed/CommonManaged.props]: Why did you remove the stable version suffix workaround?

Comment (eerhardt) [reply src/managed/CommonManaged.props]: Because this is handled by the root dir.props now.

https://github.com/dotnet/core-setup/blob/bb6dac8404513d5ac5a1a216e362ec2ff5ea8ef5/dir.props#L74-L76

Technically, this change should be made in `master` as well.

Comment (weshaggard) [reply src/managed/CommonManaged.props]: I see. Interestingly it doesn't show up when I do "view all" diff it only shows up on the individual commit which is a little odd.

Comment (eerhardt) [reply src/managed/CommonManaged.props]: That's because this addition came in a cherry-picked commit, and then I removed it as part of "fixing up changes from master".  Originally this was needed because the "managed" projects didn't import the root dir.props.  So this change was needed.  But then later I changed it so these projects do import the root dir.props.  That makes this code unnecessary now.

I'll port this change to master as well, since it is unnecessary there.

**Comment by eerhardt** [comment]
@weshaggard I'm going to merge this PR into dev/release/2.0.0.  The CI run in release/2.0.0 (#3246) is green.
